sql: import the go sql library as gosql

RaduBerinde opened the pull request:
When browsing code (especially testing code), the use of "sql." for the go
"database/sql" library (as opposed to our sql package) gets confusing. This
change renames all imports of "database/sql" to "gosql".

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/6434)

<!-- Reviewable:end -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, though without a `make check`, this will likely rot. 

---

Review status: 0 of 24 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6434#-:-KGhBVqyZKiU6u9_sW_I:-1988583858)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR! Good point, added a make check test.

---

Review status: 0 of 25 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6434#-:-KGijziXyMvtzpUkk3dK:106067494)*
<!-- Sent from Reviewable.io -->

RaduBerinde committed:
sql: import the go sql library as gosql

When browsing code (especially testing code), the use of "sql." for the go
"database/sql" library (as opposed to our sql package) gets confusing. This
change renames all imports of "database/sql" to "gosql".

RaduBerinde committed:
sql: add make check for database/sql import
